packages/php-wasm/compile/php/phpwasm-emscripten-library.js

```diff
@@ -354,32 +383,41 @@ const LibraryExample = {
 			});
 
 			// Pass data from child process's stdout to PHP's end of the stdout pipe.
-			const stdoutStream = SYSCALLS.getStreamFromFD(stdoutChildFd);
+			const stdoutStream = std[1]
+				? SYSCALLS.getStreamFromFD(ProcInfo.stdoutChildFd)
+				: null;
 			let stdoutAt = 0;
 			cp.stdout.on('data', function (data) {
 				ProcInfo.stdout.emit('data', data);
-				stdoutStream.stream_ops.write(
-					stdoutStream,
-					data,
-					0,
-					data.length,
-					stdoutAt
-				);
+
+				if (stdoutStream) {
+					stdoutStream.stream_ops.write(
+						stdoutStream,
+						data,
+						0,
+						data.length,
+						stdoutAt
+					);
+				}
 				stdoutAt += data.length;
 			});
```

Comment:
This entire fragment should be fine to wrap in `if(std[1])`. Ditto for stderr below.

Reply:
Or even better – `if(ProcInfo.stdoutChildFd)`